lib/routes/admin-api/feature.js

```diff
@@ -11,22 +11,27 @@ const {
 const NameExistsError = require('../../error/name-exists-error');
 const { handleErrors } = require('./util');
 const extractUser = require('../../extract-user');
+const {
+    UPDATE_FEATURE,
+    DELETE_FEATURE,
+    CREATE_FEATURE,
+} = require('../../permissions');
 const { featureShema, nameSchema } = require('./feature-schema');
 const version = 1;
 
 class FeatureController extends Controller {
-    constructor({ featureToggleStore, eventStore }) {
-        super();
-        this.featureToggleStore = featureToggleStore;
-        this.eventStore = eventStore;
+    constructor(config) {
+        super(config);
+        this.featureToggleStore = config.stores.featureToggleStore;
+        this.eventStore = config.stores.eventStore;
 
         this.get('/', this.getAllToggles);
-        this.post('/', this.createToggle);
+        this.post('/', this.createToggle, CREATE_FEATURE);
         this.get('/:featureName', this.getToggle);
-        this.put('/:featureName', this.updateToggle);
-        this.delete('/:featureName', this.deleteToggle);
+        this.put('/:featureName', this.updateToggle, UPDATE_FEATURE);
+        this.delete('/:featureName', this.deleteToggle, DELETE_FEATURE);
         this.post('/validate', this.validate);
-        this.post('/:featureName/toggle', this.toggle);
+        this.post('/:featureName/toggle', this.toggle, UPDATE_FEATURE);
```

Comment:
I was thinking on another approach where all the permission handling is taken out in a separate middleware.

Pseud-code for how this would be implemented: 

```javascript
function hasPermission(user, permission) {
    return (
        user &&
        user.permissions &&
        (user.permissions.indexOf(ADMIN) !== -1 ||
            user.permissions.indexOf(permission) !== -1)
    );
}

function permissionMiddleware(permission) {
    return (req, res, next) => {
        if (hasPermission(req.user, permission)) {
            next();
        } else {
            return res
                .status(403)
                .json(
                    new MissingPermission({
                        permission,
                        message: `You require ${permission} to perform this action`,
                    })
                )
                .end();
        }
}

function permissionify(app) {
    app.post('/api/admin/feature', permissionMiddleware(CREATE_FEATURE));
    app.put('/api/admin/feature/:featureName', permissionMiddleware(UPDATE_FEATURE));
    app.post('/api/admin/feature/:featureName', permissionMiddleware(UPDATE_FEATURE));
    app.delete('/api/admin/feature/:featureName', permissionMiddleware(UPDATE_FEATURE));
}

module.exports = permissionify;
```

We could even build `paths` and `methods` in to the permissions themselves, making the permission object holding all information. E.g.:

```javascript
FEATURE_CREATE = {
   permissions: [{path: '/api/admin/feature', method: 'POST'}]
}
```

**The pros:**
- Permission is a single concern 
- Applied only if activated
- Less code changes (I believe)

**The cons:**
- Less explicit than you suggestion 
- Easy to forget to deny access without permission. (This might be the case with your solution also? Should the default be reverted, aka: no explicit permission means no permission?)


**What do you think @Kouzukii?**

__I'm just throwing this in there, we can even consider this change as a later refactoring task.__